tests/func/test_ignore.py

```diff
@@ -56,3 +61,92 @@ def test_ignore_in_parent_dir(self):
         all_paths = self._get_all_paths()
 
         self.assertNotIn(forbidden_path, all_paths)
+
+
+def test_metadata_unchanged_when_moving_ignored_file(dvc_repo, repo_dir):
+    new_data_path = repo_dir.DATA_SUB + "_new"
+    dvcignore = DvcIgnoreFilter(root=dvc_repo.root_dir)
+
+    ignore_file = os.path.join(dvc_repo.root_dir, DvcIgnore.DVCIGNORE_FILE)
+    repo_dir.create(
+        ignore_file,
+        "\n".join(
+            [to_posixpath(repo_dir.DATA_SUB), to_posixpath(new_data_path)]
+        ),
+    )
+    dvcignore.load_upper_levels(repo_dir.DATA_DIR)
+
+    mtime_sig, size = get_mtime_and_size(repo_dir.DATA_DIR, dvcignore)
+
+    shutil.move(repo_dir.DATA_SUB, new_data_path)
+
+    new_mtime_sig, new_size = get_mtime_and_size(repo_dir.DATA_DIR, dvcignore)
+
+    assert new_mtime_sig == mtime_sig
+    assert new_size == size
+
+
+def test_mtime_changed_when_moving_non_ignored_file(repo_dir):
+    new_data_path = repo_dir.DATA_SUB + "_new"
+    mtime, size = get_mtime_and_size(repo_dir.DATA_DIR)
+
+    shutil.move(repo_dir.DATA_SUB, new_data_path)
+    new_mtime, new_size = get_mtime_and_size(repo_dir.DATA_DIR)
+
+    assert new_mtime != mtime
+    assert new_size == size
+
+
+def test_metadata_unchanged_on_ignored_file_deletion(dvc_repo, repo_dir):
+    ignore_file = os.path.join(dvc_repo.root_dir, DvcIgnore.DVCIGNORE_FILE)
+    repo_dir.create(ignore_file, to_posixpath(repo_dir.DATA_SUB))
+    dvcignore = DvcIgnoreFilter(root=dvc_repo.root_dir)
+    dvcignore.load_upper_levels(repo_dir.DATA_DIR)
+
+    mtime, size = get_mtime_and_size(repo_dir.DATA_DIR, dvcignore)
```

Comment:
Should be just `dvc_repo.dvcignore`, by making manual construction and extra calls we fail to test the actual code.

Reply:
Yes, we had a bit of discussion with @efiop about those tests. The problem is that we would need to trigger the dvcignore loading. But I guess this problem would be solved if I adjusted to your suggestions about loading dvcignore upon creating the filter.